core/src/test/java/org/infinispan/interceptors/impl/AsyncResultTest.java

```diff
@@ -0,0 +1,50 @@
+package org.infinispan.interceptors.impl;
+
+import static org.testng.AssertJUnit.assertEquals;
+
+import org.infinispan.test.AbstractInfinispanTest;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+/**
+ * @author Dan Berindei
+ * @since 9.0
+ */
+@Test(groups = "unit", testName = "interceptors.AsyncResultTest")
+public class AsyncResultTest extends AbstractInfinispanTest {
+   private static final AsyncResult.Invoker0 INVOKER0 = (callback, returnValue, throwable) -> null;
```

Comment:
There are only 2 cases that really matter here, whether `head < tail` or `head >= tail`.